File: LeetCode/Python Solutions/Squares of a Sorted Array/squares.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        result = [0] * n  # Initialize the result array with zeros
        left, right = 0, n - 1  # Two pointers
        for i in range(n - 1, -1, -1):  # Fill the result array from the end
            if abs(nums[left]) < abs(nums[right]):
                square = nums[right]
                right -= 1
            else:
                square = nums[left]
                left += 1
            result[i] = square * square
        return result

File: LeetCode/Java Solutions/Squares of a Sorted Array/squares.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];  // Initialize the result array
        int left = 0, right = n - 1;  // Two pointers
        for (int i = n - 1; i >= 0; i--) {  // Fill the result array from the end
            if (Math.abs(nums[left]) < Math.abs(nums[right])) {
                result[i] = nums[right] * nums[right];
                right--;
            } else {
                result[i] = nums[left] * nums[left];
                left++;
            }
        }
        return result;
    }
}
